### PR TITLE
22354 latestReviewComment

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#22354

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/core/filing.py [hunk 1]
```diff
@@ -53,6 +53,12 @@ class Status(str, Enum):
         PAPER_ONLY = 'PAPER_ONLY'
         PENDING_CORRECTION = 'PENDING_CORRECTION'
 
+        # filings with staff review
+        APPROVED = 'APPROVED'
+        AWAITING_REVIEW = 'AWAITING_REVIEW'
+        CHANGE_REQUESTED = 'CHANGE_REQUESTED'
+        REJECTED = 'REJECTED'
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/core/filing.py:60]: Are you sure about AWAITING_REVIEW?

File: legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py [hunk 2]
```diff
@@ -270,8 +280,13 @@ def get_single_filing(identifier: str, filing_id: int):
...
             filing_json['filing']['documents'] = documents
 
-        if filing_json.get('filing', {}).get('header', {}).get('status') == Filing.Status.PENDING.value:
+        if rv.status == Filing.Status.PENDING.value:
             ListFilingResource.get_payment_update(filing_json)
+        elif (rv.status in [Filing.Status.CHANGE_REQUESTED.value,
+                            Filing.Status.APPROVED.value,
+                            Filing.Status.REJECTED.value] and
+              (review_result := ReviewResult.get_last_review_result(rv.id))):
+            filing_json['filing']['header']['latestReviewComment'] = review_result.comments
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py:289]: 👍

**Review by severinbeauvais — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]
Looks good

File: legal-api/src/legal_api/core/filing.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/core/filing.py:60]: I did not understand

Comment (severinbeauvais) [reply legal-api/src/legal_api/core/filing.py:60]: The comment says, "filings with staff review". I am not exactly sure what this refers to. However, only Approved, Change Requested, and Rejected can have a review and comment, right?

**Review by PaulGarewal — APPROVED** [review]

File: legal-api/src/legal_api/core/filing.py (hunk 1 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/core/filing.py:60]: Oh! So a filing that is awaiting review also has a review object? In that case, yes, these 4 filing states have a review object. Resolving.

**Pull request merged**